src/v/v8_engine/script.cc

```diff
@@ -92,9 +137,31 @@ void script::set_function(std::string_view name) {
     _function.Reset(_isolate.get(), function_val.As<v8::Function>());
 }
 
-// It is the first version for run. In next updates I use alient_thread for run
-// v8.
-void script::run(ss::temporary_buffer<char>& data) { run_internal(data); }
+ss::future<> script::run(ss::temporary_buffer<char>& data) {
+    return ss::with_semaphore(_mutex, 1, [this, &data] {
```

Comment:
`data` must be moved into this lambda, then moved into `run_internal()`

Reply:
Oh, I missed this moment when refactored

Reply:
Done